site/reference/fn/sink/README.md

```diff
@@ -1,51 +1,45 @@
 ---
-title: "Sink"
+title: "`sink`"
 linkTitle: "sink"
 type: docs
 description: >
-   Specify a directory as an output sink package
+   Write resources to a local directory
 ---
 
 <!--mdtogo:Short
-    Specify a directory as an output sink package
+   Write resources to a local directory
 -->
 
-Implements a [sink function] by reading STDIN and writing configuration.
+`sink` reads resources from `stdin` and writes them to a local directory.
+It is useful for chaining functions using Unix pipe. For more details, refer to
+[Chaining functions].
 
-Sink will not prune / delete files for delete resources because it only knows
-about files for which it sees input resources.
-
-### Examples
+### Synopsis
 
-<!--mdtogo:Examples-->
+<!--mdtogo:Long-->
 
 ```shell
-# run a function using explicit sources and sinks
-kpt fn source DIR/ |
-  kpt fn run --image gcr.io/example.com/my-fn |
-  kpt fn sink DIR/
+kpt fn sink DIR [flags]
+
+DIR:
+  Path to a local directory to write resources to. Directory must exist.
 ```
 
 <!--mdtogo-->
 
-### Synopsis
+### Examples
 
-<!--mdtogo:Long-->
+<!--mdtogo:Examples-->
 
 ```shell
-kpt fn sink [DIR]
-
-DIR:
-  Path to a package directory.  Defaults to stdout if unspecified.
+# read resources from DIR directory, execute my-fn on them and write the
+# output to DIR directory.
+$ kpt fn source DIR |
+  kpt fn eval --image gcr.io/example.com/my-fn - |
+  kpt fn sink DIR
```

Comment:
All of the examples here use `source` and `sink` together. Do they always have to be used in conjunction? What happens if one is used without the other? It might be helpful to explicitly answer these questions in the documentation or have examples

Reply:
`source`, 'eval' and `sink` can be used to create your own little pipeline where `source` acts as a generator (builtin) function, followed by set of `eval` functions and then final `sink` function that writes the output. So far, that is the only use-case I know of. I want to see more use-cases of these in the field.